firedrake/slope_limiter/vertex_based_limiter.py

```diff
@@ -86,11 +81,12 @@ def compute_bounds(self, field):
         self.max_field.assign(-1.0e10)  # small number
         self.min_field.assign(1.0e10)  # big number
 
-        par_loop(self._min_max_loop,
-                 dx,
-                 {"maxq": (self.max_field, RW),
-                  "minq": (self.min_field, RW),
-                  "q": (self.centroids, READ)})
+        # FIXME: Relies on implementation detail in PyOP2
+        # since op2.RW does not work the way it was expected to
+        # we are using op2.WRITE in all cases and it happens to
+        # read and write from the buffer
```

Comment:
To elaborate a little.  When interpolating a UFL expression, we build a par_loop like so:

```python
par_loop(kernel, itset, output(op2.WRITE, map[op2.i[0]]), input_arg(op2.READ, input_map)))
```

The wrapper code allocates a buffer for the output (zeroed), executes the kernel, and then does a write back to the output dat.  If the input_args include the output dat as READ, then the value is staged in and read in the kernel (there are no write conflicts in the kernel itself because the kernel writes to a buffer).  Then the write back doesn't have conflicts because it occurs after the kernel.

If I change the access descriptor on output to `RW`, then the buffer is allocated (not zeroed, but rather the current values are staged in).  But the writeback is never executed! (I think this is a PyOP2 bug).  But I think we haven't in part thought hard enough about the semantics of such loops (especially where the dat argument appears in multiple places).